test/src/tests-serverModel.ts

```diff
@@ -0,0 +1,1677 @@
+                .getInstance()
+                ._ServerModel.createEventObject(event);
+
+            expect(sdkEvent).to.be.ok;
+            expect(sdkEvent.UserIdentities).to.eql(expectedUserIdentities);
+            expect(sdkEvent.MPID).to.equal('98765');
+            expect(sdkEvent.UserAttributes).to.eql(expectedUserAttributes);
+            expect(sdkEvent.ConsentState).to.be.ok;
+
+            done();
+        });
+
+        it('Should append identities when user present', function(done) {
+            let sdkEvent = mParticle
+                .getInstance()
+                ._ServerModel.createEventObject(event);
+
+            sdkEvent.should.be.ok;
+            expect(sdkEvent.UserIdentities).to.eql([]);
+
+            var user = {
+                getUserIdentities: () => {
+                    return {
+                        userIdentities: {
+                            customerid: '1234567',
+                            email: 'foo-email',
+                            other: 'foo-other',
+                            other2: 'foo-other2',
+                            other3: 'foo-other3',
+                            other4: 'foo-other4',
+                            not_a_valid_id: 'foo',
+                        },
+                    };
+                },
+                getAllUserAttributes: () => {
+                    return null;
+                },
+                getMPID: () => {
+                    return null;
+                },
+                getConsentState: () => {
+                    return null;
+                },
+            };
+
+            var identityMapping = {};
+            identityMapping[Types.IdentityType.CustomerId] = '1234567';
+            identityMapping[Types.IdentityType.Email] = 'foo-email';
+            identityMapping[Types.IdentityType.Other] = 'foo-other';
+            identityMapping[Types.IdentityType.Other2] = 'foo-other2';
+            identityMapping[Types.IdentityType.Other3] = 'foo-other3';
+            identityMapping[Types.IdentityType.Other4] = 'foo-other4';
+
+            mParticle.getInstance()._ServerModel.appendUserInfo(user, sdkEvent);
+            sdkEvent.UserIdentities.should.be.ok;
+            sdkEvent.UserIdentities.length.should.equal(6);
+
+            sdkEvent.UserIdentities.forEach(function(id) {
+                var type = id.Type;
+                var value = id.Identity;
+                identityMapping[type].should.equal(value);
+            });
+
+            done();
+        });
+
+        it('Should append user attributes when user present', function(done) {
+            let sdkEvent = mParticle
+                .getInstance()
+                ._ServerModel.createEventObject(event);
+
+            sdkEvent.should.be.ok;
+            expect(sdkEvent.UserAttributes).to.eql({});
+            var attributes = { foo: 'bar', 'foo-arr': ['bar1', 'bar2'] };
+            var user: MParticleUser = {
+                // TODO: Compiler is not happy with this
```

Comment:
My compiler seemed okay, is it because this TODO was not removed and you added the `as unknown` below to rectify this?

Reply:
Turns out it was a typo on line 1583